See:
https://bugs.gentoo.org/637488
https://bugs.kde.org/show_bug.cgi?id=387686

commit 2b5eab6a8db1b0487a3ad7fc4e7eeda6d3513626
Author: Mark Wielaard <mark@klomp.org>
Date:   Thu Jun 29 15:26:30 2017 +0000

    memcheck/tests: Use ucontext_t instead of struct ucontext
    
    glibc 2.26 does not expose struct ucontext anymore.
    
    Signed-off-by: Khem Raj <raj.khem@gmail.com>
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@16457

diff --git a/memcheck/tests/linux/stack_changes.c b/memcheck/tests/linux/stack_changes.c
index a978fc2b0..7f97b90a5 100644
--- a/memcheck/tests/linux/stack_changes.c
+++ b/memcheck/tests/linux/stack_changes.c
@@ -10,7 +10,7 @@
 // This test is checking the libc context calls (setcontext, etc.) and
 // checks that Valgrind notices their stack changes properly.
 
-typedef  struct ucontext  mycontext;
+typedef  ucontext_t  mycontext;
 
 mycontext ctx1, ctx2, oldc;
 int count;
